src/components/app.jsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import Axios from 'axios';
+import LocationDetails from './location-details';
+import ForecastSummaries from './forecast-summaries';
+import ForecastDetails from './forecast-details';
+import SearchForm from './search-form';
+import '../styles/app.scss';
+
+class App extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      selectedDate: 0,
+      forecasts: [],
+      location: {
+        city: '',
+        country: '',
+      },
+    };
+    this.handleForecastSelect = this.handleForecastSelect.bind(this);
+    this.handleSearchForm = this.handleSearchForm.bind(this);
+  }
+
+  handleForecastSelect(date) {
+    this.setState({
+      selectedDate: date,
+    });
+  }
+
+  handleSearchForm(city) {
+    Axios.get(`https://mcr-codes-weather.herokuapp.com/forecast?city=${city}`)
+      .then(response => {
+        this.setState({
+          forecasts: response.data.forecasts,
+          location: response.data.location,
+          selectedDate: response.data.forecasts[0].date,
+        });
+      })
+      .catch(() => {
+        alert('Cannot find city. Please try again.');
```

Comment:
Good thinking adding in the catch block. There is usually information contained in error that can help you customise the message you are giving the user though. 

```
.catch(error => {
  alert(error.message);
})
```